component/client/chat/ClientIndependentChatRoom.tsx

```diff
@@ -39,18 +44,31 @@ class ClientIndependentChatRoom extends React.Component<any, IClientIndependentC
           });
         }
       });
-    ClientIndependentChatRoom.getCurrentTabUrl((url) => {
+    ClientIndependentChatRoom.getCurrentTabUrl(this.state.currentUrl, (url) => {
       if (url) {
         this.setState({
           currentUrl: url
         });
       }
     });
+
+    this.handleUpdate = this.handleUpdate.bind(this);
+    this.unsubscribe = storage.subscribe(this.handleUpdate);
+  }
+
+  public componentWillUnmount() {
+    this.unsubscribe();
   }
 
   public render() {
     return <ClientChat url={this.state.currentUrl} username={this.state.username} userID={this.state.userID}/>;
   }
+
+  private handleUpdate(data) {
+    this.setState((prevState, props) => {
+      return {username: (data.username && data.username.newValue) ? data.username.newValue : prevState.username};
```

Comment:
If `username` is updated, propagate the change to the `ClientChat` component as well.